### PR TITLE
Added profileController

### PR DESCRIPTION
**Pull request by the-ayushgupta**
Added profileController

### CODE REVIEW

**Review by anantgupta001 — CHANGES_REQUESTED** [review]
please go thorugh the required changes

File: Server/routes/profile.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    res.status(400).send("User is not logged in!");
+}
+// Middleware to check if the user is logged in
+router.put(
+    '/perosnalInfo/editProfile',
+    ensureAuthenticated,
+    profileController.editProfile
+);
+router.delete(
+    '/perosnalInfo/deleteProfile',
```
Comment (anantgupta001) [inline Server/routes/profile.js:19]: set the correct route for this.

File: Server/routes/profile.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    ensureAuthenticated,
+    profileController.editProfile
+);
+router.delete(
+    '/perosnalInfo/deleteProfile', 
+    ensureAuthenticated,
+    profileController.deleteProfile
+);
+router.put(
+    '/perosnalInfo/likedFoods',
```
Comment (anantgupta001) [inline Server/routes/profile.js:24]: this too.

File: Server/server.js [hunk 3]
```diff
@@ -16,6 +16,8 @@ const User = require("./models/user.js");
 const messmenuRouter = require("./routes/messmenu.js");
 const homeRouter = require('./routes/home');
 const authRouter = require("./routes/auth.js");
+const editProfileRouter = require("./routes/profile.js");
```
Comment (anantgupta001) [inline Server/server.js:19]: name the variable correctly.

File: Server/controllers/block.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        if (!blocks || blocks.length === 0) {
+            return { status: false, message: 'No blocks found' };
+        }
+        return { status: true, blocks: blocks };
+    } catch (err) {
+        return { status: false, message: 'Server error', error: err.message };
+    }
+}
+
+module.exports.show = (req, res) => {
```
Comment (anantgupta001) [inline Server/controllers/block.js:19]: name the function accordingly

**Review by anantgupta001 — CHANGES_REQUESTED** [review]
issues must be resolved before merging the PR

**Review by anantgupta001 — CHANGES_REQUESTED** [review]
Issues must be resolved before merging this PR

**Review by anantgupta001 — APPROVED** [review]
LGTM

**Review by ImmaNOOBie — APPROVED** [review]

**Pull request merged**